sentry-core/src/client.rs

```diff
@@ -275,6 +279,40 @@ impl Client {
         Default::default()
     }
 
+    /// Assembles an [`Event`] and [`SessionUpdate`].
+    ///
+    /// Adds the additional context provided in the specified [`Scope`]
+    /// just as when using [`capture_event`](crate::Client::capture_event), but **without** actually sending
+    /// the event to sentry.
+    pub fn assemble_event(
```

Comment:
After checking with other SDK developers, the conclusion was that this proposed fn does not fit well into the unified API specs. Since this is independent from implementing attachment support, please remove this for now along with the changes to `prepare_event`.

Reply:
Do you have a suggestion on how I can create an event when a crash occurs so I can serialize it to disk with all of the relevant information on the current `Scope`?

Reply:
Specifically due to the stated requirement in the Envelopes documentation

> For minidump and apple crash report attachments, the corresponding "event" Item must be sent within the same Envelope.

Reply:
I would have said to create a custom transport that can dump envelopes to disk, but that might be a bit complex to do.
In native we basically do the same, but we create the envelope in the minidump callback (= in the signal handler).